diff --git a/buildsystem/compile-libvlc.sh b/buildsystem/compile-libvlc.sh
index 58aeed2..651134c 100755
--- a/buildsystem/compile-libvlc.sh
+++ b/buildsystem/compile-libvlc.sh
@@ -543,7 +543,7 @@ if [ ${ANDROID_API} -lt "21" ] ; then
     # force uselocale using libandroid_support since it's present in libc++
     export ac_cv_func_uselocale=yes
 
-    VLC_LDFLAGS="-landroid_support"
+    VLC_LDFLAGS="-landroid_support -lc++_static"
 fi
 
 # always use fixups for search.h and tdestroy
@@ -677,7 +677,7 @@ echo -e "ndk-build vlc"
 touch $VLC_OUT_PATH/dummy.cpp
 
 $NDK_BUILD -C $LIBVLCJNI_SRC_DIR/libvlc \
-    APP_STL="c++_shared" \
+    APP_STL="c++_static" \
     APP_CPPFLAGS="-frtti -fexceptions" \
     VLC_SRC_DIR="$VLC_SRC_DIR" \
     VLC_BUILD_DIR="$VLC_BUILD_DIR" \
