diff --git a/buildsystem/compile-libvlc.sh b/buildsystem/compile-libvlc.sh
index 58aeed2..7733f52 100755
--- a/buildsystem/compile-libvlc.sh
+++ b/buildsystem/compile-libvlc.sh
@@ -533,6 +533,8 @@ mkdir -p $VLC_BUILD_DIR
 # CONFIGURE #
 #############

+VLC_LDFLAGS="-lc++_static -static-libgcc"
+
 if [ ${ANDROID_API} -lt "26" ]; then
     # android APIs < 26 have empty sys/shm.h headers that triggers shm detection but it
     # doesn't have any shm functions and/or symbols. */
@@ -543,7 +545,7 @@ if [ ${ANDROID_API} -lt "21" ] ; then
     # force uselocale using libandroid_support since it's present in libc++
     export ac_cv_func_uselocale=yes

-    VLC_LDFLAGS="-landroid_support"
+    VLC_LDFLAGS="${VLC_LDFLAGS} -landroid_support"
 fi

 # always use fixups for search.h and tdestroy
@@ -677,7 +679,7 @@ echo -e "ndk-build vlc"
 touch $VLC_OUT_PATH/dummy.cpp

 $NDK_BUILD -C $LIBVLCJNI_SRC_DIR/libvlc \
-    APP_STL="c++_shared" \
+    APP_STL="c++_static" \
     APP_CPPFLAGS="-frtti -fexceptions" \
     VLC_SRC_DIR="$VLC_SRC_DIR" \
     VLC_BUILD_DIR="$VLC_BUILD_DIR" \
